src/controllers/articleRatingControllers.js

```diff
@@ -0,0 +1,63 @@
+import model from '../db/models/index';
+
+const { Users, Articles, Rating } = model;
+
+/**
+ * rating controller
+ */
+class ArticleManager {
+  /**
+     *
+     * @param {Object} req
+     * @param {Object} res
+     * @returns {Object} return object
+     */
+  static async rateArticle(req, res) {
+    const { slug } = req.params;
+    const { rating } = req.body;
+
+    const findArticle = await Articles.findOne({ where: { slug } });
+    if (!findArticle) {
+      return res.status(404).send({
+        error: 'Article not found!'
+      });
+    }
+    const findUser = await Users.findOne({ where: { email: req.user.email } });
+
+    if (findUser.id === findArticle.postedBy) {
+      return res.status(400).json({
+        error: 'You can not rate your article!'
+      });
+    }
+
+    const findIfUserRatedArticle = await Rating.findOne({
+      where: {
+        user: findUser.id,
+        articleId: findArticle.id
+      }
+    });
+    if (findIfUserRatedArticle) {
+      return res.status(400).json({
+        error: 'You have already rated this article'
+      });
+    }
+
+    const articleRatings = await Rating.findOne({ where: { articleId: findArticle.id, rating } });
```

Comment:
what's the purpose of this line?

Reply:
This line helps in finding a user rating an article before, with the same rate value requested

Reply:
Don't think this check is necessary, since you have already blocked the user from rating again.